pkg/cloudprovider/provider/alibaba/provider.go

```diff
@@ -278,7 +278,13 @@ func (p *provider) Cleanup(machine *v1alpha1.Machine, data *cloudprovidertypes.P
 		return false, fmt.Errorf("failed to delete instance with instanceID %s, due to %v", c.InstanceID, err)
 	}
 
-	return false, nil
+	if err := data.Update(machine, func(updatedMachine *v1alpha1.Machine) {
+		updatedMachine.Finalizers = kuberneteshelper.RemoveFinalizer(updatedMachine.Finalizers, finalizerInstance)
```

Comment:
I'm not entirely sure, but probably we can omit second RemoveFinalizer() call (and logic), and simply return the error, to let this method be called once again.

Reply:
Yeah actually this is how it was. But when I changed it to true I duplicated it. Maybe its better to revert to my old implementation, even though alicoud delete instances sync.